docs/code/VBufferCareFeeding.md

```diff
@@ -142,129 +136,130 @@ using (IRowCursor cursor = dataview.GetRowCursor(col => col == 5))
 }
 ```
 
-In this example, we open a cursor (telling it to make only column 5 active),
-then get the "getter" over this column. What enables buffer re-use for this is
-that, as we go row by row over the data with the `while` loop, we pass in the
-same `value` variable in to the `getter` delegate, again and again. Presumably
-the first time, or several, memory is allocated. Initially `value =
-default(VBuffer<float>)`, that is, it has zero `Length` and `Count` and `null`
-`Indices` and `Values`. Presumably at some point, probably the first call,
-`value` is replaced with a `VBuffer<float>` that has actual values allocated.
-In subsequent calls, perhaps these are judged as insufficiently large, and new
-arrays are allocated, but we would expect at some point the arrays would
+To be explicit, a `ValueGetter` is defined this way:
+
+```csharp
+public delegate void ValueGetter<TValue>(ref TValue value);
+```
+
+Anyway, in this example, we open a cursor (telling it to make only column 5
+active), then get the "getter" over this column. What enables buffer re-use
+for this is that, as we go row by row over the data with the `while` loop, we
+pass in the same `value` variable in to the `getter` delegate, again and
+again. Presumably the first time, or several, memory is allocated (though,
+internally, and invisibly to us). Initially `VBuffer<float> value = default`,
+that is, it has zero `Length` and similarly empty spans. Presumably at some
+point, probably the first call, `value` is replaced with a `VBuffer<float>`
+that has actual values, stored in freshly allocated buffers. In subsequent
+calls, *perhaps* these are judged as insufficiently large, and new arrays are
+internally allocated, but we would expect at some point the arrays would
 become "large enough" to accommodate many values, so reallocations would
-become increasingly rare.
+become increasingly rare. In this way, we eventually avoid nearly all
```

Comment:
`we eventually avoid nearly all allocations and garbage collection.`

=>

`we avoid future buffer allocations and require less garbage collection.`

?

Reply:
Well... I sort of consider eventually a more accurate statement, because if you think about sparse vectors, or variable size vectors, every time you have a value that exceeds the current allocation you must reallocate, so it's not like you avoid *future* allocations right away, you merely reach a state where the buffer has reached what is likely its maximum size and you become increasingly unlikely to need a reallocation, but that is something that happens eventually. So my word choice here was deliberate. But... I see your point it's a suitably subtle point that your simpler language will probably be best, so I will change.

Reply:
Maybe we can also rework the previous sentence and tie this whole thing together. The point I was thinking is that: `at some point the arrays would become "large enough", and we would no longer have to allocate new buffers after that point`. That is a true statement - at some point we've reached the max it will need, and no longer have to allocate.

Reply:
OK. It's also a bit more compact. I also added an additional phrase to also mention garbage collection, since that's *really* the thing we used to observe killing us in terms of perf.